### PR TITLE
If no documents processed in reindex, do not fail

### LINKED ISSUE #1170: Reindex failed. Error - The index or alias identified by <dest_index> was not found
### Expected Behavior
Reindex operation should have been successfully completed

### Actual Behavior

Reindex failed. The index or alias identified by <dest_index> was not found

### Steps to Reproduce the Problem
**curator-reindex script**
_actions:  
  1:  
    description: "Reindex index1 into index2"
    action: reindex
    options:
      disable_action: False
      wait_interval: 9
      max_wait: -1
      request_body:
        source:
          index: test-events
        dest:
          index: reindex_test-events
    filters:
    - filtertype:_ 

**Curator Logs**
2018-03-15 14:38:12,821 INFO      Preparing Action ID: 1, "reindex"
2018-03-15 14:38:12,828 INFO      Trying Action ID: 1, "reindex": Reindex index1 into index2
2018-03-15 14:38:12,845 INFO      Commencing reindex operation
2018-03-15 14:38:12,851 INFO      Task "reindex from [test-events] to [reindex_test-events]" with task_id "7aroLMAqTeyG6u-_f7N_cQ:45155" has been running for 0.003328736 seconds
2018-03-15 14:38:21,868 INFO      Task "reindex from [test-events] to [reindex_test-events]" completed at 2018-03-15T14:38:12Z.
2018-03-15 14:38:21,871 ERROR     The index described as "reindex_test-events" was not found after the reindex operation. Check Elasticsearch logs for more information.
2018-03-15 14:38:21,871 ERROR     Failed to complete action: reindex.  <class 'curator.exceptions.FailedExecution'>: Exception encountered.  Rerun with loglevel DEBUG and/or check Elasticsearch logs for more information. Exception: Reindex failed. The index or alias identified by "reindex_test-events" was not found.

### Specifications

  - Version:5.4.1/5.4.0
  - Platform: ElasticSearch 6.2

## Context (Environment)
We can't do reindexing of restored index.

### PR DESCRIPTION
**Pull request by afharo**
Fixes #1170 

## Proposed Changes

  - If no documents are processed while reindexing (total number of documents == 0), don't check if the target index exists (because it won't be created).

N.B.: Sorry if my syntax is not correct. I'm actually a NodeJS developer and tried my best with python. Please, feel free to update the code if necessary :)

### CODE REVIEW

**Comment by afharo** [comment]
I've already signed the the CLA. How can I reset the check?

**Comment by untergeek** [comment]
I see how it works, and what you're trying to do, but I'm not sure I like how it's assembled.

There's something about this doesn't sit completely right with me, and I can't put my finger on it yet.

I will think about this for a while. I'm AFK this next week, so my response might be delayed.

**Comment by afharo** [comment]
Thank you for having a look at it this fast.

I agree: I’m sure there’s a better way:
- like adding a filter that checks the number of documents returned by a query 
- or simply performing the query before calling the reindex
- or performing the same query after the reindex to know how many documents can be found in the source
- or performing the “total” check only if the options have with source.query (although, if no query performed but an empty index found, it will fail anyway)
- or making this behavior optional (I don’t think it’s a good idea because I’d say this is a bug, not just a feature)
- or any other logic you can think of
- or simply a better implementation of my approach xD 

Let me know if I can help you with anything.

**Comment by afharo** [comment]
Hi @untergeek,
Any chance you've been able to have a look at this?

**Comment by untergeek** [comment]
Sorry for the long wait. I've been all over the world since May 13 (Orlando, FL 2x; Tokyo; Osaka; Austin, TX; San Diego, CA; Melbourne, AUS), and that's put a definite damper on my ability to keep up with PRs and such.  I'm barely beginning to be able to catch up.

I'm looking now. I'll see what I find.

File: curator/actions.py [hunk 1]
```diff
@@ -1307,28 +1307,64 @@ def _get_reindex_args(self, source, dest):
...
+        This function calls client.tasks.get with the provided `task_id`.  It will get the value 
+        from ``'response.total'`` as the total number of elements processed during reindexing.
+        If the value is not found, it will return -1
+
+        :arg task_id: A task_id which ostensibly matches a task searchable in the
+            tasks API.
+        """
+        try:
+            task_data = self.client.tasks.get(task_id=task_id)
+        except Exception as e:
```
Comment (untergeek) [inline curator/actions.py]: Lines 1321 and 1322 are not tested. A mock exception raised in a unit test would satisfy this.

**Comment by afharo** [comment]
@untergeek thank you for your comments. I just pushed the changes to tackle them. Please, let me know if I can help on anything else.

**Comment by afharo** [comment]
When is 5.7.7 planned to be published?

**Comment by untergeek** [comment]
I'm finally working from home this week, and plan on it probably being a 5.8.0 release. Barring weirdness, sometime this week is my goal.

**Pull request merged**